src/lib/OAuth2/OAuth2User.ts

```diff
@@ -0,0 +1,71 @@
+import { User } from 'oidc-client';
+import { AuthorizationTypes } from 'shared/constants';
+import { PropertiesOf } from 'shared/types';
+
+export interface IOAuth2User {
```

Comment:
This is a wrapper for the `User` class from the `oidc-client` module.
This holds useful information as tokens, groups, and expiration date.